Dockerfile

```
@@ -14,9 +16,11 @@ RUN cross_debian_arch=$(uname -m | sed -e 's/aarch64/amd64/'  -e 's/x86_64/arm64
     apt-get autoremove --yes
 
 COPY go.mod /tmp/go.mod
+
 # Extract Go version from go.mod
 RUN GO_VERSION=$(grep -oPm1 '^go \K([[:digit:].]+)' /tmp/go.mod) && \
-    wget -qO- https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz | tar -C /usr/local -xz
+    wget -qO- https://golang.org/dl/go${GO_VERSION}.linux-${GOARCH}.tar.gz | tar -C /usr/local -xz
```

```suggestion
    wget -qO- https://golang.org/dl/go${GO_VERSION}.linux-$(uname -m | sed -e 's/aarch64/arm64/' -e 's/x86_64/amd64/').tar.gz | tar -C /usr/local -xz
```

Similar to https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/220/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L7 this could just use `uname -m` so there is no need for external argument here.

---

Thanks, pushed